### PR TITLE
Add plane/box and plane/sphere intersection test

### PR DESCRIPTION
**Pull request by Mugen87**
I've implemented two common and handy intersection test:
- intersection test between `THREE.Plane`and `THREE.Sphere`
- intersection test between `THREE.Plane`and `THREE.Box3`

### CODE REVIEW

**Comment by mrdoob** [comment]
Nice!

**Comment by mrdoob** [comment]
Thanks!

File: src/math/Sphere.js [hunk 1]
```diff
@@ -104,6 +104,20 @@ THREE.Sphere.prototype = {
...
+		// We use the following equation to compute the signed distance from
+		// the center of the sphere to the plane.
+		//
+		// distance = q * n - d
+		//
+		// If this distance is greater than the radius of the sphere,
+		// then there is no intersection.
+
+		return ( ( this.center.dot( plane.normal ) - plane.constant ) <= this.radius );
+
```
Comment (WestLangley) [inline src/math/Sphere.js]: Shouldn't you be taking the absolute value of the signed distance?

Do you mind adding unit tests to demonstrate the correctness of the two new methods?

Comment (Mugen87) [inline src/math/Sphere.js]: @WestLangley You're right, i've forgot to implement the absolute value of the distance! 

Oh, i didn't realize that there are unit tests for three.js! Sry, i will add some tests...

**Comment by yomotsu** [comment]
Do we need to add such as intersection tests in three.js?
I've seen that three.js is a library for actual rendering, it was long time ago tho
see: https://github.com/mrdoob/three.js/issues/49

If so, I can add some more futures for intersection tests as well as collision detection.
https://github.com/yomotsu/meshwalk.js/blob/gh-pages/src/core/collision.js

**Comment by yomotsu** [comment]
I forgot to mention... @mrdoob @WestLangley
wondering if you could see..
https://github.com/mrdoob/three.js/pull/7438#issuecomment-151005693

**Comment by Mugen87** [comment]
I think it's advantageous for three.js, if at least the most fundamental geometric tests with common primitives (Ray, AABB, Sphere, Plane etc.) are present. Even the simplest interactive applications often needs such logic....

**Comment by mrdoob** [comment]
Yes, but I would try to keep them as simple as possible.

**Comment by yomotsu** [comment]
I understand, thanks! :+1: 

BTW, should we rename the Box3 and Box2 methods called `isIntersectionBox` to `intersectsBox` as well?
https://github.com/mrdoob/three.js/blob/master/src/math/Box3.js#L234
https://github.com/mrdoob/three.js/blob/master/src/math/Box2.js#L168

**Comment by mrdoob** [comment]
Already done in the `dev` branch 😊

**Comment by yomotsu** [comment]
cool! thanks!

**Pull request merged**